libelement/src/common_internal.cpp

```diff
@@ -156,9 +156,10 @@ std::string ast_to_string(const element_ast* ast, int depth, const element_ast*
     return string;
 }
 
-std::string instruction_to_string(const element::instruction& expression, std::size_t depth)
+std::string instruction_to_string(const element::instruction& expression, std::size_t depth, std::string prefix)
 {
     std::string string(depth, ' ');
+    string += prefix;
```

Comment:
Not sure if in scope, but... I feel like this function is doing enough string concatenation that it might be a good idea to use `std::stringstream` instead to avoid massive reallocation overheads.

Reply:
agreed not to improve performance here right now, as it serves as a debugging tool for which performance doesn't matter that much